backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/orchestration/impl/ElasticsearchDefaultWorkSequenceBuilder.java

```diff
@@ -33,10 +32,8 @@
 	private final Supplier<ContextualErrorHandler> errorHandlerSupplier;
 	private final BulkResultExtractionStepImpl bulkResultExtractionStep = new BulkResultExtractionStepImpl();
 
-	private CompletableFuture<Void> refreshFuture;
-	private CompletableFuture<?> sequenceFuture;
-	private ElasticsearchRefreshableWorkExecutionContext executionContext;
-	private ContextualErrorHandler errorHandler;
+	private CompletableFuture<?> currentlyBuildingSequenceTail;
```

Comment:
This field is not final even within a sequence. That's why I called it "tail" and not "starter": we regularly update its value to reflect the latest work added to the sequence.

I'd rather not move it to the sequence context, because this particular object is only ever updated by the builder, while we build the sequence. There is no reason to update it, or even read it, in any lambda.

I agree that this code is still quite confusing, though. At some point we'll have to change it a bit, maybe split the logic a bit more; but I don't know how to do this right now, especially not if we want to avoid additional object instances. I'll keep that for later.